src/emit/substrate/target.rs

```diff
@@ -1051,19 +1051,104 @@ impl<'a> TargetRuntime<'a> for SubstrateTarget {
         (data, length)
     }
 
-    fn print(&self, binary: &Binary, string_ptr: PointerValue, string_len: IntValue) {
+    fn print(
+        &self,
+        binary: &Binary,
+        string_ptr: PointerValue,
+        string_len: IntValue,
+        source: PrintSource,
```

Comment:
This might be abusing the `print` instruction facility. I thought it quite nice when print basically just called into seal_debug and that's it. Also this way we have code duplication for other targets?

On a high level, it seems to be the better solution to leave the formatting responsibility to the callers inside codegen. @seanyoung WDYT?

Reply:
I also think this is a mistake. I have another version of formatting print in codegen instead of emit and it looks much more better

Reply:
With that said, I will push a commit soon with print formatting done in codegen